include/pmacc/math/vector/TwistComponents.hpp

```diff
@@ -70,12 +69,12 @@ namespace pmacc
          * @return reference of the input vector with twisted axes.
          */
         template<typename T_Axes, typename T_Vector>
-        HDINLINE typename result_of::TwistComponents<T_Axes, T_Vector>::type twistComponents(T_Vector& vector)
+        HDINLINE auto twistComponents(T_Vector& vector)
         {
             /* The reinterpret_cast is valid because the target type is the same as the
              * input type except its navigator policy which does not occupy any memory though.
              */
-            return reinterpret_cast<typename result_of::TwistComponents<T_Axes, T_Vector>::type>(vector);
+            return reinterpret_cast<typename detail::TwistComponents<T_Axes, T_Vector>::type>(vector);
```

Comment:
Is the `typename` keyword still required here? (Does not hurt anyways, I am just curious)

Reply:
Yes because of `X::type`